addons/mail/controllers/discuss.py

```diff
@@ -1,18 +1,221 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-import base64
-import werkzeug.utils
-import werkzeug.wrappers
+import json
 
 from odoo import http
-from odoo.exceptions import AccessError
 from odoo.http import request
+from odoo.tools import consteq
+from odoo.tools.misc import get_lang
+from odoo.tools.translate import _
+from werkzeug.exceptions import NotFound
 
 
 class DiscussController(http.Controller):
 
-    @http.route('/mail/read_followers', type='json', auth='user')
+    # --------------------------------------------------------------------------
+    # Public Pages
+    # --------------------------------------------------------------------------
+
+    @http.route('/chat/<int:channel_id>/<string:invitation_token>', methods=['GET'], type='http', auth='public')
```

Comment:
shouldn't this be better namespaced?

```suggestion
    @http.route('/chat/join/<int:channel_id>/<string:invitation_token>', methods=['GET'], type='http', auth='public')
```

Reply:
The goal is to have the shortest share URL possible. Note that I'm expecting this URL might change again soon depending on feedback that will inevitably come when the oxp videos will be made and people start to test it more seriously.